bin/strata-client/src/extractor.rs

```diff
@@ -54,22 +53,7 @@ pub(super) async fn extract_deposit_requests<L1DB: L1Database>(
         .map_err(RpcServerError::Db)?;
 
     let deposit_info_iter = l1_txs.into_iter().filter_map(move |l1_tx| {
-        let mut raw_tx = l1_tx.tx_data();
-        let tx = Transaction::consensus_decode(&mut raw_tx);
-        if tx.is_err() {
```

Comment:
So for this we could instead a expose a `.try_parse_tx()` fn that returns a `Result<Transaction, Error>`.

Unrelated but I'm still perplexed as to why this module is in this crate, I know it was explained to me once but it's still wrong.

Reply:
`try_into` probably makes more sense actually, even if it's more implicit.